src/pokemon.c

```diff
@@ -4878,24 +4877,18 @@ bool8 PokemonUseItemEffects(struct Pokemon *mon, u16 item, u8 partyIndex, u8 mov
                 retVal = FALSE;
             }
 
-            // Rare Candy
+            // Rare Candy / EXP Candy
             if ((itemEffect[i] & ITEM3_LEVEL_UP)
              && GetMonData(mon, MON_DATA_LEVEL, NULL) != MAX_LEVEL)
             {
-                if (ItemId_GetHoldEffectParam(item) & FULL_LEVEL)
+                u8 param = ItemId_GetHoldEffectParam(item);
+                if (param == 0xFF) // Rare Candy
                 {
                     dataUnsigned = gExperienceTables[gBaseStats[GetMonData(mon, MON_DATA_SPECIES, NULL)].growthRate][GetMonData(mon, MON_DATA_LEVEL, NULL) + 1];
                 }
-                else
+                else if (param < NUM_EXP_CANDY_SIZES) // EXP Candies
```

Comment:
`NUM_EXP_CANDY_SIZES` could be replaced by `ARRAY_COUNT(sExpCandyExperienceTable)`